src/backend/cdb/motion/ic_tcp.c

```diff
@@ -1775,7 +1775,7 @@ SetupTCPInterconnect(EState *estate)
 		i = 0;
 		while (n > 0 &&
 			   outgoing_count < expectedTotalOutgoing &&
-			   i < sendingChunkTransportState->numConns)
+			   i < sendingChunkTransportState ? sendingChunkTransportState->numConns : 0)
```

Comment:
I'm not sure about the sequence of operations here. Maybe we should add explicit parentheses?